### PR TITLE
Allow DRF PictureField to accept aspect_ratios

### PR DESCRIPTION
**Pull request by amureki**
You may provide optional GET parameters to the serializer, to specify the aspect
ratio and breakpoints you want to include in the response. The parameters are
prefixed with the `fieldname_` to avoid conflicts with other fields.

```bash
curl http://localhost:8000/api/path/?picture_ratio=16%2F9&picture_m=6&picture_l=4
# %2F is the url encoded slash
```

```json
{
  "other_fields": "…",
  "picture": {
    "url": "/path/to/image.jpg",
    "width": 800,
    "height": 800,
    "ratios": {
      "1/1": {
        "sources": {
          "image/webp": {
            "100": "/path/to/image/1/100w.webp",
            "200": "…"
          }
        },
        "media": "(min-width: 0px) and (max-width: 991px) 100vw, (min-width: 992px) and (max-width: 1199px) 33vw, 25vw"
      }
    }
  }
}
```

Note that the `media` keys are only included, if you have specified breakpoints.

### CODE REVIEW

**Comment by codingjoe** [comment]
@amureki I added a commit that replicates the same API from the picture template tag. I have a feeling that might be more consistent. We could also consider having those attributes passed via GET parameters, so you can have a more dynamic REST API. Thoughts?

**Comment by amureki** [comment]
@codingjoe yeah, that looks better. I'd be fine moving forward with it!

File: tests/contrib/test_rest_framework.py [hunk 1]
```diff
@@ -97,3 +97,28 @@ def test_to_representation(self, image_upload_file, settings):
...
+        settings.PICTURES["USE_PLACEHOLDERS"] = False
+
+        profile = models.Profile.objects.create(picture=image_upload_file)
+        serializer = ProfileSerializer(profile)
+        serializer.fields["picture"].ratio = "16/9"
+        serializer.fields["picture"].breakpoints = {"m": 4, "l": 3}
+
+        assert serializer.data["picture"] == {
+            "sources": {
+                "WEBP": {
```
Comment (codingjoe) [inline tests/contrib/test_rest_framework.py:112]: I wonder if this shouldn't be the MIME-type. I'd need it anyway in the HTML. That way, you wouldn't even need to know what is being served.

**Comment by codingjoe** [comment]
We could even build a React and Vue component for the package, or at least an example, right?

File: tests/contrib/test_rest_framework.py (hunk 1 above)
Comment (codingjoe) [reply tests/contrib/test_rest_framework.py:112]: OK, how would you limit the result set? Wasn't that kinda the problem?

Comment (codingjoe) [reply tests/contrib/test_rest_framework.py:112]: Media only works, if you provide some kind of grid info. It's also identical to all sources.

**Comment by codingjoe** [comment]
@amureki I pushed an updates, that includes your suggestions. I am curious what you think.

**Comment by amureki** [comment]
> @amureki I pushed an updates, that includes your suggestions. I am curious what you think.

Yeah, that's good and interesting idea with the `GET` params. I wonder if we should prefix those somehow to not clash with other possible API params that other dev could define in their APIs...

File: tests/contrib/test_rest_framework.py (hunk 1 above)
Comment (amureki) [reply tests/contrib/test_rest_framework.py]: But that's already a case - in your last commit, payload could be reduced by not using all aspect ratios. I don't know how we could approach `sources` as I am lacking understanding the need of so many variants...

**Comment by codingjoe** [comment]
> > @amureki I pushed an updates, that includes your suggestions. I am curious what you think.
> 
> Yeah, that's good and interesting idea with the `GET` params. I wonder if we should prefix those somehow to not clash with other possible API params that other dev could define in their APIs...

Good idea, we should at least add the field name a prefix in case there are multiple picture fields. Do you think it'll be sufficient.

**Comment by codingjoe** [comment]
@chgad, we are proposing some changes to the DRF field. Since you suggested the original implementation. How do you feel about this?

**Comment by codingjoe** [comment]
cc @SebastianKapunkt

**Review by codingjoe — APPROVED** [review]
I am OK with this now, but I'm waiting for feedback from current users.

**Comment by SebastianKapunkt** [comment]
It helps to reduce the payload of the request. To get the actual source you want to use in the frontend, you still need a JS snipped to extract what you need.

Maybe it makes sense to even add more parameters to the request, to the point where you only get one URL back.

**Comment by codingjoe** [comment]
> Maybe it makes sense to even add more parameters to the request, to the point where you only get one URL back.

@SebastianKapunkt why would you want only one URL? You mean, if you already know all device information like pixel density and the final size? Pixel density is known per device, but even apps can be resized (e.g. via screen rotation). So, you'd never know the actual size.

> To get the actual source you want to use in the frontend, you still need a JS snipped to extract what you need.

We wanted to keep the API payload as atomic as possible. Having a single `srcset`-string would be helpful in HTML only. That being said, a simple `Array.join(', ')` should be feasible for most.

@amureki maybe we should consider adding examples for React and Vue components to the docs?

**Comment by amureki** [comment]
>@amureki maybe we should consider adding examples for React and Vue components to the docs?

#56 is a good start. Then we could add some code examples for React and Vue to the docs as well. Should we also move the docs to readthedocs then?

**Pull request merged**